caluma/caluma_form/structure.py

```diff
@@ -140,35 +140,60 @@ def fields(self):
         return self._fields
 
     @property
-    def sub_forms(self):
+    def sub_forms(self) -> List[Field]:
         if self._sub_forms is None:
-            self._sub_forms = {
-                field.question.slug: field
+            self._sub_forms = [
+                field
                 for field in self.children()
                 if field.question.type == Question.TYPE_FORM
-            }
+            ] + [
+                child
+                for field in self.children()
+                for child in field.children()
+                if field.question.type == Question.TYPE_TABLE
+            ]
         return self._sub_forms
 
-    def get_field(
-        self, question_slug: str, check_parent: bool = True
-    ) -> Optional[Field]:
+    def get_fields(self, question_slug: str, check_parent: bool = True) -> List[Field]:
+        """Collect fields where the question occurs throughout this structure.
+
+        Cases:
+        0. question not in structure
+        1. question is in the same form (-> greedily returns the question)
+        2. question in a neighbor form, ie. answer would be in same document (excluding tables)
+        3. question in multiple neighbor forms
+        4. question in a table form (same fieldset)
+        5. (question in a table form (different row))
+        6. question in a table form, lower than current fieldset
+        7. question in upper structure (from table row)
+
+        Expected:
+        0: return []
+        1-3: answer exists once, but might be in multiple forms -> multiple fields
+        4: return only row-local fields (not looking up / other rows)
+        5: incomplete row missing answer -> return fields with empty value (like case 4)
+        6: return all fields for all rows
+        7: same as 1-3
+        """
 
         field = self.fields.get(question_slug)
 
-        if not field and check_parent:
-            field = self.parent().get_field(question_slug) if self.parent() else None
         if field:
-            return field
+            return [field]
+        if not field and check_parent:
```

Comment:
The `not field and` part is redundant here. If `field` exists, we would have returned already in the line above

Reply:
done